integrations/server_test.go

```diff
@@ -0,0 +1,519 @@
+				if err != nil {
+					t.Fatalf("failed to add server: %v", err)
+					return
+				}
+				tt.fields.Servers[i] = s
+			}
+
+			// Populate Layouts
+			for i, layout := range tt.fields.Layouts {
+				l, err := boltdb.LayoutsStore.Add(ctx, layout)
+				if err != nil {
+					t.Fatalf("failed to add layout: %v", err)
+					return
+				}
+				tt.fields.Layouts[i] = l
+			}
+
+			// Populate Dashboards
+			for i, dashboard := range tt.fields.Dashboards {
+				d, err := boltdb.DashboardsStore.Add(ctx, dashboard)
+				if err != nil {
+					t.Fatalf("failed to add dashboard: %v", err)
+					return
+				}
+				tt.fields.Dashboards[i] = d
+			}
+
+			_ = boltdb.Close()
+
+			go tt.args.server.Serve(ctx)
+			serverURL := fmt.Sprintf("http://%v:%v%v", host, port, tt.args.path)
+
+			timeout := time.Now().Add(100 * time.Millisecond)
+			for {
+				_, err := http.Get(serverURL + "/swagger.json")
+				if err == nil {
+					break
+				}
+				if time.Now().After(timeout) {
+					t.Fatalf("failed to start server")
+					return
+				}
+			}
+
+			// Construct HTTP Request
+			buf, _ := json.Marshal(tt.args.payload)
+			reqBody := ioutil.NopCloser(bytes.NewReader(buf))
+			req, _ := http.NewRequest(tt.args.method, serverURL, reqBody)
+			token, _ := oauth2.NewJWT(tt.args.server.TokenSecret).Create(ctx, tt.args.principal)
+			req.AddCookie(&http.Cookie{
+				Name:     "session",
+				Value:    string(token),
+				HttpOnly: true,
+				Path:     "/",
+			})
+
+			// Make actual http request
+			resp, err := http.DefaultClient.Do(req)
+			if err != nil {
+				t.Fatalf("failed to make httprequest: %v", err)
+				return
+			}
+
+			content := resp.Header.Get("Content-Type")
+			body, _ := ioutil.ReadAll(resp.Body)
+
+			if resp.StatusCode != tt.wants.statusCode {
+				t.Errorf(
+					"%s %s Status Code = %v, want %v",
+					tt.args.method,
+					tt.args.path,
+					resp.StatusCode,
+					tt.wants.statusCode,
+				)
+			}
+
+			if tt.wants.contentType != "" && content != tt.wants.contentType {
+				t.Errorf(
+					"%s %s Content Type = %v, want %v",
+					tt.args.method,
+					tt.args.path,
+					content,
+					tt.wants.contentType,
+				)
+			}
+
+			if eq, err := jsonEqual(tt.wants.body, string(body)); err != nil || !eq {
+				t.Errorf(
+					"%s %s Body = %v, want %v",
+					tt.args.method,
+					tt.args.path,
+					string(body),
+					tt.wants.body,
+				)
+			}
+
+			tt.args.server.Listener.Close()
+		})
+	}
+}
+
+func hostAndPort() (string, int) {
+	s := httptest.NewServer(nil)
+	defer s.Close()
+
+	u, err := url.Parse(s.URL)
+	if err != nil {
+		panic(err)
+	}
+	xs := strings.Split(u.Host, ":")
+	host := xs[0]
+	portStr := xs[1]
+	port, err := strconv.Atoi(portStr)
+	if err != nil {
+		panic(err)
+	}
+	return host, port
+
+}
+
+func newBoltFile() string {
+	f, err := ioutil.TempFile("", "chronograf-bolt-")
+	if err != nil {
+		panic(err)
+	}
+	f.Close()
+
+	return f.Name()
+}
+
+func jsonEqual(s1, s2 string) (eq bool, err error) {
```

Comment:
can this be imported from where it was already created before? or move that common function to a utils file?

Reply:
If it was a more complex function I'd pull it out somewhere common. I was thinking about making this a general JSON diffing thing, but I haven't gotten around to it.

In general I'm not a huge fan of util packages.